grakn-graph/src/test/java/ai/grakn/graph/property/RelationTypePropertyTest.java

```diff
@@ -0,0 +1,270 @@
+import static ai.grakn.graph.property.PropertyUtil.directSubTypes;
+import static ai.grakn.util.ErrorMessage.META_TYPE_IMMUTABLE;
+import static ai.grakn.util.ErrorMessage.SUPER_TYPE_LOOP_DETECTED;
+import static ai.grakn.util.Schema.MetaSchema.isMetaName;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeThat;
+
+/**
+ * @author Felix Chapman
+ */
+@RunWith(JUnitQuickcheck.class)
+public class RelationTypePropertyTest {
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Property
+    public void whenDeletingMetaRelationType_Throw(@Open GraknGraph graph) {
+        RelationType relation = graph.admin().getMetaRelationType();
+
+        exception.expect(ConceptException.class);
+        exception.expectMessage(META_TYPE_IMMUTABLE.getMessage(relation.getName()));
+
+        relation.delete();
+    }
+
+    @Property
+    public void whenANonMetaRelationTypeHasNoInstancesSubTypesOrRules_ItCanBeDeleted(
+            @Open GraknGraph graph, @FromGraph @Meta(false) RelationType type) {
+        assumeThat(type.instances(), empty());
+        assumeThat(type.subTypes(), contains(type));
+        assumeThat(type.getRulesOfHypothesis(), empty());
+        assumeThat(type.getRulesOfConclusion(), empty());
+
+        type.delete();
+
+        assertNull(graph.getType(type.getName()));
+    }
+
+    @Property
+    public void whenAddingARelationOfAMetaType_Throw(@Meta RelationType type) {
+        exception.expect(ConceptException.class);
+        exception.expectMessage(META_TYPE_IMMUTABLE.getMessage(type.getName()));
+        type.addRelation();
+    }
+
+    @Property
+    public void whenAddingARelation_TheDirectTypeOfTheRelationIsTheTypeItWasCreatedFrom(
+            @Meta(false) RelationType type) {
+        Relation relation = type.addRelation();
+
+        assertEquals(type, relation.type());
+    }
+
+    @Property
+    public void whenAddingARelation_TheRelationIsInNoRelations(@Meta(false) RelationType type) {
+        Relation relation = type.addRelation();
+
+        assertThat(relation.relations(), empty());
+    }
+
+    @Property
+    public void whenAddingARelation_TheRelationHasNoResources(@Meta(false) RelationType type) {
+        Relation relation = type.addRelation();
+
+        assertThat(relation.resources(), empty());
+    }
+
+    @Property
+    public void whenAddingARelation_TheRelationHasNoRolePlayers(@Meta(false) RelationType type) {
+        Relation relation = type.addRelation();
+
+        assertThat(relation.rolePlayers(), empty());
+    }
+
+    @Property
+    public void whenSettingTheDirectSuperTypeOfAMetaType_Throw(
```

Comment:
The issue here is the setter for `superType` is not on the `Type` interface and takes different parameters. I can look into writing a generic test, but I don't know how elegant it will be.